lib/skill-matcher.sh

```
@@ -0,0 +1,151 @@
+
+validate_inputs
+
+echo "Finding skills for: \"$TASK_DESC\""
+echo ""
+
+# Convert task to lowercase for matching
+task_lower=$(echo "$TASK_DESC" | tr '[:upper:]' '[:lower:]')
+
+# Create temp file for skill data (bash 3.2 compatible - no associative arrays)
+TMPFILE=$(mktemp)
+trap "rm -f $TMPFILE" EXIT
```

Shellcheck (0.11.0)

[warning] Use single quotes, otherwise this expands now rather than when signalled.

(SC2064)